### PR TITLE
#3867 Rename attribute WindowsFact-> WindowsFactAttribute

### LINKED ISSUE #3867: Rename WindowsFact to WindowsFactAttribute to follow best practices
https://github.com/cake-build/cake/blob/fdb906acf82fb3210836caa50c441323e7fcc22b/src/Cake.Testing.Xunit/WindowsFact.cs#L11

### PR DESCRIPTION
**Pull request by ivanbibik**
Fixes #3867
Rename attribute `WindowsFact`->`WindowsFactAttribute`.

Or I can create a new `WindowsFactAttribute` attribute and `WindowsFact` attribute mark [Obsolete]

Thank you in advance

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/cake-build/cake?pullRequest=3962) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/cake-build/cake?pullRequest=3962) before we can accept your contribution.<br/><br/>:x: ivanbibik [sign now](https://cla.dotnetfoundation.org/cake-build/cake?pullRequest=3962)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/cake-build/cake?pullRequest=3962) it.</sub>

File: src/Cake.Testing.Xunit/WindowsFact.cs [hunk 1]
```diff
@@ -8,17 +8,17 @@
 
 namespace Cake.Testing.Xunit
 {
-    public sealed class WindowsFact : FactAttribute
+    public sealed class WindowsFactAttribute : FactAttribute
```
Comment (devlead) [inline src/Cake.Testing.Xunit/WindowsFact.cs:11]: This is the only change needed (+ rename file), compiler will add `Attribute` automatically,

**Review by devlead — CHANGES_REQUESTED** [review]
Left some feedback.
This is a breaking change so this won't be merged until the next major version.

File: src/Cake.Testing.Xunit/WindowsFact.cs (hunk 1 above)
Comment (ivanbibik) [reply src/Cake.Testing.Xunit/WindowsFact.cs:11]: Thanks, done

**Review by augustoproiete — APPROVED** [review]
LGTM

**Comment by devlead** [comment]
@ivanbibik something went wrong in the rebase of this PR, I'm sorry, but could you please reopen PR and I'll review it?

**Pull request closed without merging**